python/revng/api/manager.py

```diff
@@ -117,7 +117,13 @@ def _produce_target(
             # TODO: we really should be able to provide a detailed error here
             raise RevngException("Failed to produce targets")
 
-        return {t.serialize(): t.extract() for t in targets}
+        result = {}
+        for target in targets:
+            extracted_target = target.extract()
+            if extracted_target is None:
+                raise RevngException("Target extraction failed")
```

Comment:
Can we have a more informative error message?
Maybe add the target name.

What would help you if you were to read an error message about this?

Reply:
Added the serialized target to the exception message